client/src/language/languageClient.ts

```diff
@@ -61,10 +61,10 @@ export async function activateLanguageServer (context: ExtensionContext): Promis
 
       // Notify the server about file changes to '.clientrc files contain in the workspace
       fileEvents: [
-        workspace.createFileSystemWatcher('**/*.bbclass', false, true, false),
-        workspace.createFileSystemWatcher('**/*.inc', false, true, false),
-        workspace.createFileSystemWatcher('**/*.bb', false, true, false),
-        workspace.createFileSystemWatcher('**/*.conf', false, true, false)
+        workspace.createFileSystemWatcher('**/*.bbclass', false, false, false),
+        workspace.createFileSystemWatcher('**/*.inc', false, false, false),
+        workspace.createFileSystemWatcher('**/*.bb', false, false, false),
+        workspace.createFileSystemWatcher('**/*.conf', false, false, false)
```

Comment:
This feature should be configurable and also not run if settings are not sane. Look for the mecanisms for this in server.ts on file save and make sure they are applied to watched changes. Could reuse the same settings?

Reply:
The `onDidSave` currently also triggers `parseAllRecipes`, and usually saving a file will now trigger the change event (because I changed the boolean value to `false`). Hence the multiple parses as I mentioned in another conversation. I would prefer to remove the code in `onDidSave`.

As for the settings, do we want to have a general setting for all files or one for each file type? Also, in case you don't know, those 3 boolean values in the `createFileSystemWatcher` are respectively `createEvent`, `changeEvent` and `deleteEvent`. These may all be configurable.

Reply:
Bitbake files can only be modified if:
 - the user edits them (so he will save them and we already start parse)
 - the user runs devtool commands (we already rescan just what's necessary)
So I think we should plainly remove this feature. There's no external bitbake modfications that require any action as I'm aware of.

Reply:
Some possible modifications are coming from `git`. Which was the main reason I created the ticket in the redmine

Reply:
if we don't care about `git`, then we can remove the feature and close this PR

Reply:
`onDidSave` is very usefull to get bitbake diagnostics quickly. Since scanning is so massive, I think we should remove the watchedFiles features. I probed other internal users and they unanimously agreed.

Reply:
Then we might want to remove the `parseAllRecipes` from `connection.onDidChangeWatchedFiles` for now. The modification of the rest needs Io's attention

Reply:
Yes we should remove any kind of Watched Files features :+1:

Reply:
> Some possible modifications are coming from `git`. Which was the main reason I created the ticket in the redmine

If it matters, non-opened files could also change with "rename" and "Replace All" (I personally use this one all the time). I would not be surprised if there are other ways.

"EmbeddedLanguageDocsManager" use "workspace.onDidRenameFiles" and "workspace.onDidDeleteFiles" in order to avoid keeping data of non-existing files and have a memory leak. That being said, I assume the user would have to create, delete and rename millions of documents before it becomes noticeable. So I guess we can do without it.